src/plugins/mf_priority.cpp

```diff
@@ -807,6 +841,51 @@ static int depend_cb (flux_plugin_t *p,
 }
 
 
+/*
+ * When a job gets to the job.state.sched state, pack a timestamp to be
+ * associated with the job in the case that it is eligible to be run but
+ * cannot (e.g due to resource constraints). The timestamp will be used
+ * to calculate the "age" of a job and increase its priority according to
+ * the time it has been waiting to run.
+ */
+static int sched_cb (flux_plugin_t *p,
+                     const char *topic,
+                     flux_plugin_arg_t *args,
+                     void *data)
+{
+    double t_released = 0.0;
+    int urgency = 0;
+    flux_t *h = flux_jobtap_get_flux (p);
+
+    // unpack urgency
+    if (flux_plugin_arg_unpack (args,
+                                FLUX_PLUGIN_ARG_IN,
+                                "{s:i}",
+                                "urgency", &urgency) < 0)
+        return flux_jobtap_error (p, args, "unable to unpack urgency");
+
+    if (urgency > 0) {
+        // the job is not being held, and therefore we need to get the current
+        // time to keep track of how long the job has been released and waiting
+        t_released = std::chrono::duration_cast <std::chrono::duration<double>> (
+            std::chrono::system_clock::now ().time_since_epoch ()).count ();
+        // store job id, t_released in map
+        released_jobs[FLUX_JOBTAP_CURRENT_JOB] = t_released;
```

Comment:
`FLUX_JOBTAP_CURRENT_JOB` is a constant, is that really what you want to use here? I could be misunderstanding use of the map, but I think with multiple jobs you'd end up overwriting each job's `t_released`?

Seems like you could keep a map of released jobs by jobid, then you wouldn't need to set the value in the aux item list of the job, or you can allocate space for `t_released` and just store it in the job's aux item list without the need for the `released_jobs` global?

Reply:
Sorry about that, it's my own misunderstanding of `FLUX_JOBTAP_CURRENT_JOB` - I thought that I could use that in place of unpacking the jobid.

And you are absolutely right about not setting anything in the aux item list of the job if I use a map. Originally, I had tried allocating space for `t_released` instead of using a map but wasn't able to figure it out - probably because I was doing something wrong and don't understand memory allocation fully.

Can I allocate space for a `double*` in one callback and clean it up in another? In this case, allocate in `job.state.sched` and clean it up in `job.state.inactive`?

Reply:
There may be an easier way using a C++ smart pointer, but in C I'd do something like:
```c++
    // Allocate space for a double, e.g. in `job.state.sched`
    double *t_released = malloc (sizeof (double));
    if (!t_released)
        // handle enomem error
   *t_released = value;

    // Place value in job_aux hash and schedule destruction with job
    // (be sure to check return value in actual code)
    flux_jobtap_job_aux_set (p,
                           FLUX_JOBTAP_CURRENT_JOB,
                           "mf_priority::t_released",
                           t_released,
                           free);
```

I think the above will work, but untested so there may be typos, etc.

Then you'd access the value the way you're currently doing it via `flux_jobtap_job_aux_get(3)`. The value will automatically be freed when the job is destroyed, so you don't have to manually do it.

Reply:
I tried playing around with this implementation for a bit and was just never able to get it working. For now, I've pushed up a fix to just store/retrieve job IDs and their associated timestamps with a map.

I think the nice thing with this method is the ability to return the information stored in this map with the `plugin.query` callback. I'm definitely open to more discussion, though.

Reply:
EDIT: wait, I'm sorry, I think I might have just figured it out. 😆 I don't think I was correctly managing memory with my C++ pointers, but if I declare a C pointer (like you suggested), I think this actually becomes pretty straightforward. I can push up an additional commit that uses this approach of `aux_set` for comparison.

Reply:
OK! In that case, I'll leave my proposal as is. 👍 But it is good to know that we have the `aux_set`/`aux_get` implementation for the future!

Reply:
This seems reasonable.  FWIW, using a smart pointer in the aux array would kinda defeat the purpose, I could make something for us that makes it easy to capture smart pointers into something managed by the aux array, so things like lambdas or classes with complex state would be easier to manage, but doing it directly wouldn't really buy much unless you had a smart pointer already for some reason.